src/ClientFactory.php

```diff
@@ -80,7 +72,11 @@ public function create(array $args = [])
                     // Merge defaults in when not present.
                     $args[$key] = $a['default'];
                 } elseif (!empty($a['required'])) {
-                    throw new IAE("{$key} is a required client setting");
+                    // Allows custom error messages for missing values.
+                    $message = method_exists($this, "missing_{$key}")
+                        ? call_user_func([$this, "missing_{$key}"], $args)
```

Comment:
Could also do `$this->{"missing_{$key}"}($args)`. I usually avoid `call_user_func*` when I can. It doesn't really matter much though, so whatever you prefer is fine.